programs/marginfi/src/state/marginfi_account.rs

```diff
@@ -750,7 +761,10 @@ assert_struct_align!(Balance, 8);
 pub struct Balance {
     pub active: bool,
     pub bank_pk: Pubkey,
-    pub _pad0: [u8; 7],
+    /// Inherited from the bank when the position is first created and CANNOT BE CHANGED after that.
+    /// Note that all balances created before the addition of this feature use `ASSET_TAG_DEFAULT`
+    pub bank_asset_tag: u8,
```

Comment:
Depends on how we want to handle legacy compatibility. This locks all CURRENT user accounts out of using this feature, forcing users to create a new account, which is intentional/preferable. Imagine a user has deposited 100 SOL and has no other positions. This feature launches, their tag is `ASSET_TAG_DEFAULT`. The SOL bank updates to `ASSET_TAG_SOL`, but their position does not, so that account cannot deposit staked collateral.

On the FE, it also allows you see if this a staked or classic account at a glance when loading the user's profile without a second rpc call to load the bank.

Reply:
We now need this tag to correctly process remaining accounts per-position (when loading a user_account, staked positions take 4 remaining accounts, while normal positions take 2)